### DISCUSSION TITLE
Postgres migrations doesn't apply automatically when starting local development environment

### DISCUSSION THREAD

**Discussion by niksumeiko** [nhost/nhost#2560] [Q&A] [answered]
I just tried to ask a colleague to boot our project locally and migrations were not installed automatically. Even we have them committed to the repository (in `nhost/migrations/default/` dir).

By default, we don't have any tables under the `schema.public`, but the first migration shall create it after booting the project for the first time. 

Here's an example of our very first migration:
```sql
# nhost/migrations/default/1708719143789_create_problems/up.sql
CREATE TABLE public.problems (
  id UUID DEFAULT gen_random_uuid() NOT NULL,
  body TEXT NOT NULL,
  created_at TIMESTAMPTZ DEFAULT now() NOT NULL,
  PRIMARY KEY (id)
);
```

Here are the steps taken:

1. Downloading and installing Nhost CLI via https://docs.nhost.io/development/cli/getting-started
2. `nhost init`
3. `nhost up`

All services started, but the following info has been logged in to console:
```bash
Applying migrations...
INFO nothing to apply on database: default
```

When opening Dashboard > Database > schema.public:
<img width="523" alt="Screenshot 2024-02-26 at 12 18 27" src="https://github.com/nhost/nhost/assets/175385/e9c5b95e-1f66-438b-8119-33e46aba0e84">

#### Trying fresh setup myself
When I try to simulate the fresh setup flow myself, the same happens. Here's what I do to simulate it:
- Open Dashboard > Database > schema.public and delete all tables manually.
- `nhost down`
- Remove our project directory
- Clone it via git
- `cd ./my-project`
- `npm install; npm install --prefix functions`
- `nhost init`
- `nhost up`

All services start successfully, but migrations are not applied > no tables under `schema.public`, unfortunately: 
```bash
Applying migrations...
INFO nothing to apply on database: default
```

Not sure what to do, but I need a fully working local environment for peers, too.

**Comment by niksumeiko** [comment]
Maybe already applied migrations are stored somewhere outside of the project dir and I have to delete that record? 🤷‍♂️

**Reply by niksumeiko** [reply]
I have found that migration information is stored somewhere within Hasura. According to the [Hasura docs](https://hasura.io/docs/latest/migrations-metadata-seeds/manage-migrations/#check-the-status-of-migrations), the `hasura migrate status` could list existing migrations.

But I can't find a way to access it via the Nhost CLI.

**Reply by dbarrosop** [reply, marked as answer]
Migrations are always applied. If it says there is nothing to apply that might mean they have already been applied. If needed, you can "reset" your local environment with `nhost down --volumes && rm -rf .nhost`. You can also check your migrations with the following command:

```
nhost dev hasura migrate status --endpoint https://local.hasura.nhost.run --admin-secret nhost-admin-secret --database-name default
```

(you may need to change the admin secret)

If you still don't find the issue, please, provide a github repository reproducing the issue so we can take a look.

**Reply by niksumeiko** [reply]
The `nhost down --volumes && rm -rf .nhost` did the job. 

Now after `nhost up`, the following was printed inside the console:
```
Applying migrations...
INFO migrations applied on database: default
```
